packages/default-theme/src/logic/useDomains.js

```diff
@@ -1,34 +1,69 @@
 import { computed } from "@vue/composition-api"
-import { getApplicationContext } from "@shopware-pwa/composables"
+import {
+  getApplicationContext,
+  useCms,
+  useNotifications,
+} from "@shopware-pwa/composables"
+import { getSeoUrls } from "@shopware-pwa/shopware-6-client"
+
+// the "last-chance route always has a name starting with `all_` - nuxt default"
+const PAGE_RESOLVER_ROUTE_PREFIX = "all_"
 
 export const useDomains = (rootContext) => {
-  const { router, routing } = getApplicationContext(rootContext, "useDomains")
+  const { router, routing, route, apiInstance } = getApplicationContext(
+    rootContext,
+    "useDomains"
+  )
 
   const availableDomains = computed(() => routing.availableDomains || [])
   const currentDomainId = computed(() => routing.getCurrentDomain().domainId)
+  const trimDomain = (url) => url.replace(routing.getCurrentDomain().url, "")
+  const getPathWithoutDomain = () => trimDomain(rootContext.$route.fullPath)
+
+  const isRouteStatic = () => !route.name.startsWith(PAGE_RESOLVER_ROUTE_PREFIX)
+  const getNewDomainUrl = async (domain) => {
+    let url = `${domain.url !== "/" ? `${domain.url}` : ""}`
+    let path = ""
+
+    if (isRouteStatic()) {
+      path += `/${getPathWithoutDomain()}`.replace(/\/\/+/, "/")
+    } else {
+      const { categoryId } = useCms(rootContext)
+      try {
+        // find the correspoding URL for current page if it comes from page resolver - dynamically generated
+        const seoResponse = await getSeoUrls(
+          categoryId.value,
+          domain.languageId,
+          apiInstance
+        )
+        if (seoResponse.length && seoResponse[0].seoPathInfo) {
+          path += seoResponse[0].seoPathInfo
+        }
```

Comment:
Fallback when no SEO URL is found - go to either `/detail/${foreignKey}` or `/navigation/${foreignKey}`.

Important: As discussed, use constants instead of `/detail` and `/navigation`, because these are used in other places as well.

Reply:
the else statemend does that, it takes the URL from packages/helpers/src/cms/getCmsTechnicalPath.ts, there are a constant prefix either for navigation or for detail

Reply:
I swear, that else part wasn't there when I commented.






Could be wrong though :D